crates/networking/p2p/rlpx/connection.rs

```diff
@@ -159,49 +195,83 @@ impl<S: AsyncWrite + AsyncRead + std::marker::Unpin> RLPxConnection<S> {
         match &self.state {
             RLPxConnectionState::Established(_) => {
                 info!("Started peer main loop");
+                // Wait for eth status message or timeout.
+                let _ = timeout(Duration::from_secs(15), self.wait_status_msg())
+                    .await
+                    .map_err(|_err| {
+                        RLPxError::HandshakeError("Timeout waiting for status msg".to_string())
+                    })?;
+                let mut broadcaster_receive = self.connection_broadcast_send.subscribe();
+                // Status message received, start listening for connections,
+                // and subscribe this connection to the broadcasting.
```

Comment:
While I see the goal here, there is still a problem (I think): Status (as well as Ping, Pong and Disconnect) messages can still be received (and acted upon) inside the loop.

Reply:
Maybe we need a new State (eg. `StatusStablished`) or a flag to determine if an `eth` capability protocol was stablished, and do not brodcast transactions if that's not the case.